pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -294,7 +310,7 @@ func (e *EndpointController) onServiceUpdate(obj interface{}) {
 	}
 
 	_ = e.serviceSelectorCache.Update(key, obj.(*v1.Service).Spec.Selector)
-	e.queue.Add(key)
+	e.queueServiceSync(key, false /* allowRateLimiting */)
```

Comment:
why disallow rate limiting?

Reply:
The main use case I'm trying to optimize is to reduce number of endpoint updates during rolling update of pods in given service. For this reason, the initial version of this mechanism (the --endpoint-updates-batch-period flag) was rate limiting only pod-triggered updates. We don't get any additional value in rate limiting service-triggered updates while the user will see higher latency of updates.

Reply:
It was natural to me, but this comment clearly shows it's not obvious.
Please add a comment explaining why you do this (roughly what you wrote above).

Reply:
Added a comment